vendor: bump Pebble to 560ed0b8a18c

jbowens opened the pull request:
```
560ed0b compaction: Don't do a move compaction within L0.
b822482 compaction: Exclude largest key when extending L0 compactions if it's sentinel
585d322 internal/metamorphic: add checkpointOp, compactOp and flushOp
5bdf31a cmd/pebble: enable L0-sublevel compactions and flush splits
d478834 db: switch compaction concurrency signal from level score to L0 read-amp
c577a10 db: force flush memtables with range deletions after delay
fd3f2e3 db: add memtable count, duration and byte rate to flush event
730159f db: tweak the compaction scoring heuristics
b34dbe9 db: tighten merging iterator's seek conditions
c08345d tool: Allow vertical scrolling if visualization overflows
747113e db: fix race in TestTableStats
90dbbfc *: {white,black}list -> {allow,deny}list
```

Supersedes #50604.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50630)
<!-- Reviewable:end -->

itsbilal reviewed (approved):
Ah right, it’s the go mod days now!

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50630#-:-MAcNzsW1JwxJpPUqWfk:b-87pt88)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

jbowens commented on go.mod:
```diff
@@ -30,7 +30,7 @@ require (
 	github.com/cenkalti/backoff v2.1.1+incompatible
 	github.com/client9/misspell v0.3.4
-	github.com/cockroachdb/apd v1.1.0 // indirect
+	github.com/cockroachdb/apd v1.1.0
```
not sure why this became a direct dependency...

otan replied:
make buildshort again. You have leftover generated files that import apd

petermattis reviewed (commented):
You seem to have bumped the `apd` dependency. Is there a way to only bump the `pebble` dependency? Cc @otan

jbowens committed:
vendor: bump Pebble to 560ed0b8a18c

560ed0b compaction: Don't do a move compaction within L0.
b822482 compaction: Exclude largest key when extending L0 compactions if it's sentinel
585d322 internal/metamorphic: add checkpointOp, compactOp and flushOp
5bdf31a cmd/pebble: enable L0-sublevel compactions and flush splits
d478834 db: switch compaction concurrency signal from level score to L0 read-amp
c577a10 db: force flush memtables with range deletions after delay
fd3f2e3 db: add memtable count, duration and byte rate to flush event
730159f db: tweak the compaction scoring heuristics
b34dbe9 db: tighten merging iterator's seek conditions
c08345d tool: Allow vertical scrolling if visualization overflows
747113e db: fix race in TestTableStats
90dbbfc *: {white,black}list -> {allow,deny}list

Release note: None

otan commented on go.sum:
```diff
@@ -147,6 +147,8 @@ github.com/cockroachdb/logtags v0.0.0-20190617123548-eb05cc24525f h1:o/kfcElHqOi
 github.com/cockroachdb/logtags v0.0.0-20190617123548-eb05cc24525f/go.mod h1:i/u985jwjWRlyHXQbwatDASoW0RMlZ/3i9yJHE2xLkI=
 github.com/cockroachdb/pebble v0.0.0-20200618174744-bc7bb07c0a28 h1:OGzpqbaARCNI/VvrKMydw22zClKW3FX2fKDanLlbbXA=
 github.com/cockroachdb/pebble v0.0.0-20200618174744-bc7bb07c0a28/go.mod h1:crLnbSFbwAcQNs9FPfI1avHb5BqVgqZcr4r+IzpJ5FM=
```
i think you also need to run a `go mod tidy` to delete the old items in this file.

jbowens commented:
@otan Thanks for your help, and for making the switch to go modules!

jbowens commented:
TFTRs!

bors r+

petermattis reviewed (approved):
LGTM
